### PR TITLE
Add api change section to UpdateRC2toGA doc

### PR DESCRIPTION
**Pull request by davidkline-ms**
Adds an api changes section and describes the new changes for clearing spatial observations.

Also fixes some spelling / typing errors in the assembly name section

### CODE REVIEW

File: Documentation/UpdatingToGA.md [hunk 1]
```diff
@@ -16,8 +44,13 @@ In some instances, multiple assemblies have been merged to create better unity o
 
 The following tables describe how the RC2 .asmdef file names map to the GA release. All assembly names match the .asmdef file name. 
 
+- [MixedRealityToolkit](#mixedrealitytoolkit)
+- [MixeddRealityToolkit.Providers](#mixedrealitytoolkit.providers)
```
Comment (wiwei) [inline Documentation/UpdatingToGA.md]: Misspelled: MixeddRealityToolkit -> MixedRealityToolkit

Here and below.

File: Documentation/UpdatingToGA.md [hunk 2]
```diff
@@ -2,8 +2,36 @@
...
+
+The following method(s) have been renamed to better describe their usage.
+
+- IMixedRealitySpatialAwarenessSystem.CreateSpatialObjectParent has been renaned to IMixedRealitySpatialAwarenessSystem.CreateSpatialAwarenessObservationParent to clarify its usage.
+
+**Additions**
+
+Based on customer feedback, support for easy removal of previously observed spataial awareness data has been added.
+
+- IMixedRealitySpatialAwarenessSystem.ClearObservations()
```
Comment (wiwei) [inline Documentation/UpdatingToGA.md]: Nit: Add a link to the xref here (or code)

File: Documentation/UpdatingToGA.md [hunk 3]
```diff
@@ -2,8 +2,36 @@
...
 Between the RC2 and GA releases of the Microsoft Mixed Reality Toolkit, changes were made that may impact existing projects. This document describes those changes and how to update projects to the GA release.
 
+- [API changes](#api-changes)
 - [Assembly name changes](#assembly-name-changes)
 
+## API changes
+
+Since the release of RC2, there have been a number of API changes including some that may break existing projects. The following sections describe the changes that have occurred between the RC2 and GA releases.
+
+- [Breaking changes](#breaking-changes)
```
Comment (wiwei) [inline Documentation/UpdatingToGA.md]: I think that by definition, all of these changes in this document are breaking changes. It's not just the API stuff that's breaking.

I would just delete the "breaking changes" line and header down below, as it seems to suggest the others AREN'T breaking (not true)

File: Documentation/UpdatingToGA.md (hunk 1 above)
Comment (davidkline-ms) [reply Documentation/UpdatingToGA.md]: removing these as the doc tool apparently is not fond of bookmarks with a '.' in them. (this was the failure in doc generation: "illegal link")

**Pull request merged**